lib/auth/auth.go

```diff
@@ -1301,6 +1312,32 @@ func (a *Server) updateVersionMetrics() {
 	}
 }
 
+// updateInstallMethodsMetrics leverages the inventory control stream to report the install methods
+// of all instances that are connected to a single auth server via prometheus metrics.
+// To get an accurate representation of install methods in an entire cluster the metric must be aggregated
+// with all auth instances.
+func (a *Server) updateInstallMethodsMetrics() {
+	installMethodCount := make(map[string]int)
+
+	// record install methods for all connected resources
+	a.inventory.Iter(func(handle inventory.UpstreamHandle) {
+		installMethod := "unknown"
+
+		if len(handle.AgentMetadata().InstallMethods) > 0 {
+			slices.Sort(handle.AgentMetadata().InstallMethods)
```

Comment:
Is there any concern in sorting the install methods of the upstream handle object itself? i.e. does this modify the source object, potentially altering the inventory unintentionally?

Reply:
I'll create a copy just to be safe :+1: